### PR TITLE
Notebook Storage in S3

### PR DESCRIPTION
**Pull request by vgmartinez**
Storage the notebook in s3, is necesary set the enviroment variable AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY.

In s3 create a folder structure as follows:
bucket/youUserName/notebook/

In zeppelin-env.sh set the bucket name and userName...

### CODE REVIEW

File: conf/zeppelin-env.sh.template [hunk 1]
```diff
@@ -26,6 +26,8 @@
 # export ZEPPELIN_LOG_DIR        # Where log files are stored.  PWD by default.
 # export ZEPPELIN_PID_DIR        # The pid files are stored. /tmp by default.
 # export ZEPPELIN_NOTEBOOK_DIR   # Where notebook saved
+# export ZEPPELIN_BUCKET         # Bucket where notebook saved
+# export ZEPPELIN_USER           # User in bucket where notebook saved. For example bucket/user/notebook/2A94M5J1Z/note.json
```
Comment (Leemoonsoo) [inline conf/zeppelin-env.sh.template]: Thanks for the contribution.
The name 'ZEPPELIN_BUCKET` and `ZEPPELIN_USER` little bit ambiguous to me.
How about having some prefix for S3NotebookRepo?

For example, `ZEPPELIN_NOTEBOOK_S3_BUCKET`, `ZEPPELIN_NOTEBOOK_S3_USER`, or any better name.

**Comment by vgmartinez** [comment]
hi @Leemoonsoo,
I renamed the variables ... any suggestions are welcome...

thanks

**Comment by jersub** [comment]
Hi, thanks a lot for bringing this feature in! It maps to https://issues.apache.org/jira/browse/ZEPPELIN-118

FYI some days ago I quickly tried to achieve this by adding the `vfs-s3` lib (https://github.com/abashev/vfs-s3) to the `spark` pom. Unfortunately I couldn't manage to make it work, maybe I missed something or the lib is buggy.

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/conf/ZeppelinConfiguration.java [hunk 2]
```diff
@@ -411,6 +420,8 @@ public String getConfDir() {
     ZEPPELIN_INTERPRETER_CONNECT_TIMEOUT("zeppelin.interpreter.connect.timeout", 30000),
     ZEPPELIN_ENCODING("zeppelin.encoding", "UTF-8"),
     ZEPPELIN_NOTEBOOK_DIR("zeppelin.notebook.dir", "notebook"),
+    ZEPPELIN_NOTEBOOK_S3_BUCKET("zeppelin.bucket.name", "zeppelin"),
+    ZEPPELIN_NOTEBOOK_S3_USER("zeppelin.notebook.user", "user"),
```
Comment (Leemoonsoo) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/conf/ZeppelinConfiguration.java]: Thanks for reflecting feedback about env variable name. How about have similar prefix for Property name?
'zeppelin.bucket.name' and 'zeppelin.notebook.user' ?

**Comment by vgmartinez** [comment]
Hi @Leemoonsoo ,
The changes are ready...

thks

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/conf/ZeppelinConfiguration.java [hunk 3]
```diff
@@ -411,6 +420,8 @@ public String getConfDir() {
     ZEPPELIN_INTERPRETER_CONNECT_TIMEOUT("zeppelin.interpreter.connect.timeout", 30000),
     ZEPPELIN_ENCODING("zeppelin.encoding", "UTF-8"),
     ZEPPELIN_NOTEBOOK_DIR("zeppelin.notebook.dir", "notebook"),
+    ZEPPELIN_NOTEBOOK_S3_BUCKET("zeppelin.bucket.name", "zeppelin"),
+    ZEPPELIN_NOTEBOOK_S3_USER("zeppelin.bucket.user", "user"),
```
Comment (Leemoonsoo) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/conf/ZeppelinConfiguration.java]: Thanks for the updating the name. but can we have little bit of consistency between name of env variable and name of property? to me "zeppelin.notebook.s3.user" is more consistent. Is there special reason you chose this name?

File: conf/zeppelin-site.xml.template [hunk 4]
```diff
@@ -53,12 +53,30 @@
...
   <name>zeppelin.notebook.storage</name>
   <value>org.apache.zeppelin.notebook.repo.VFSNotebookRepo</value>
   <description>notebook persistence layer implementation</description>
 </property>
 
 <property>
+  <name>zeppelin.notebook.storage</name>
+  <value>org.apache.zeppelin.notebook.repo.S3NotebookRepo</value>
+  <description>notebook persistence layer implementation</description>
+</property>
```
Comment (Leemoonsoo) [inline conf/zeppelin-site.xml.template]: I think it's better not have two entries for the `zeppelin.notebook.storage` property at the same time.
How about put all S3NotebookRepo related configurations inside of single commented block `<!-- ... -->`  with some description.

Then user can still use this template without any modification, also user can easily understand what is necessary to setup S3NotebookRepo.

File: conf/zeppelin-site.xml.template [hunk 5]
```diff
@@ -53,12 +53,30 @@
...
 <property>
+  <name>zeppelin.bucket.name</name>
+  <value>zeppelin</value>
+  <description>path or URI for notebook persist</description>
+</property>
+
+<property>
+  <name>zeppelin.bucket.user</name>
+  <value>user</value>
+  <description>path or URI for notebook persist</description>
```
Comment (Leemoonsoo) [inline conf/zeppelin-site.xml.template]: I think the description need to be corrected.

**Comment by vgmartinez** [comment]
Hi @Leemoonsoo ,
It makes sense....I've added the comments.

thks

**Comment by vgmartinez** [comment]
Hi @Leemoonsoo ,
It makes sense....;) 
I've added the comments....

any suggestions are welcome...
thks

**Comment by Leemoonsoo** [comment]
I've tested and looks good to me.

@vgmartinez Thanks for great contribution!

**Pull request closed without merging**